contracts/src/v0.8/dev/automation/2_1/KeeperRegistry2_1.sol

```diff
@@ -461,42 +448,52 @@ contract KeeperRegistry2_1 is KeeperRegistryBase2_1, OCR2Abstract, Chainable, ER
    */
   function _validateBlockTrigger(
     uint256 upkeepId,
-    BlockTrigger memory trigger,
+    bytes memory rawTrigger,
     Upkeep memory upkeep
   ) internal returns (bool) {
-    if (trigger.blockNum < upkeep.lastPerformed) {
+    BlockTrigger memory trigger = abi.decode(rawTrigger, (BlockTrigger));
+    if (trigger.blockNum < upkeep.lastPerformedBlockNumberOrTimestamp) {
       // Can happen when another report performed this upkeep after this report was generated
-      emit StaleUpkeepReport(upkeepId);
+      emit StaleUpkeepReport(upkeepId, rawTrigger);
       return false;
     }
-    if (_blockHash(trigger.blockNum) != trigger.blockHash) {
+    if (_blockHash(trigger.blockNum) != trigger.blockHash || trigger.blockNum >= block.number) {
```

Comment:
not sure if this would work if we want to skip reorg protection (both blocknum=0, blockhash=0) but can figure that out in a followup

Reply:
yeah this is dumb. let's address in a follow up though.